### PR TITLE
Update bitly API from 3 to 4

### PR DESCRIPTION
**Pull request by Jan200101**
This PR updates the api calls to the new v4 ones and keeps existing behavior intact
the examples in the doc string have also been corrected

### CODE REVIEW

**Comment by Jan200101** [comment]
the bitly testing hasn't been updated yet I'll get that done ASAP

**Comment by Jan200101** [comment]
hmm the previously used test values no longer seem to work

File: pyshorteners/shorteners/bitly.py [hunk 1]
```diff
@@ -63,16 +69,26 @@ def expand(self, url):
         Raises:
             ExpandingErrorException: If the API Returns an error as response
         """
-        expand_url = f'{self.api_url}v3/expand'
+        self.clean_url(url)
+        if url.startswith('http'):
+            url = url.replace('https://', '').replace('http://', '')
```
Comment (Jan200101) [inline pyshorteners/shorteners/bitly.py]: currently I simply check if it starts with the protocol and remove it if needed
any suggestions for a better way to do it? I thought about regex or a dedicated function

Comment (rkagin) [reply pyshorteners/shorteners/bitly.py]: I might use urllib.parse (https://docs.python.org/3/library/urllib.parse.html).  Add `from urllib.parse import urlparse` to the top and `''.join(urlparse(url)[1:3])` here.

Comment (Jan200101) [reply pyshorteners/shorteners/bitly.py]: sounds good
I'll replace update it ASAP

**Comment by rkagin** [comment]
You'll need to also update the tests at `master/tests/test_bitly.py`, which is why this build is failing.

**Comment by Jan200101** [comment]
tests have been fixed

**Comment by rkagin** [comment]
I don't know enough about Github actions but maybe @ellisonleao can help make this merge successful?

**Comment by Jan200101** [comment]
the Github actions don't seem to work for PRs

File: pyshorteners/shorteners/bitly.py [hunk 2]
```diff
@@ -63,16 +71,21 @@ def expand(self, url):
...
-            'shortUrl': url,
-            'access_token': self.api_key,
-            'format': 'txt',
-        }
-        response = self._get(expand_url, params=params)
-        if response.ok:
-            return response.text.strip()
-        raise ExpandingErrorException(response.content)
+        self.clean_url(url)
+        url = "".join(urlparse(url)[1:3])
```
Comment (ellisonleao) [inline pyshorteners/shorteners/bitly.py]: this part doesn't sound right. Are you trying to get just the path for the url id?

File: pyshorteners/shorteners/bitly.py [hunk 3]
```diff
@@ -85,21 +98,25 @@ def total_clicks(self, url):
...
-        url = self.clean_url(url)
-        clicks_url = f'{self.api_url}v3/link/clicks'
-        params = {
-            'link': url,
-            'access_token': self.api_key,
-            'format': 'txt'
-        }
-        response = self._get(clicks_url, params=params)
+        self.clean_url(url)
+        url = "".join(urlparse(url)[1:3])
```
Comment (ellisonleao) [inline pyshorteners/shorteners/bitly.py]: same here

File: pyshorteners/shorteners/bitly.py (hunk 2 above)
Comment (Jan200101) [reply pyshorteners/shorteners/bitly.py]: yeah I used the approach mentioned [here](https://github.com/ellisonleao/pyshorteners/pull/131#discussion_r332306419)
sadly the API doesn't accept the full link anymore

Comment (ellisonleao) [reply pyshorteners/shorteners/bitly.py]: can't you just use the bit.ly id (e.g "http://bit.ly/1234" => "1234") in the url param?

Comment (Jan200101) [reply pyshorteners/shorteners/bitly.py]: no
the domain is something pro users can change so what bit.ly/1234 will expand to will not be what exampledomain.com/1234 will expand to

**Comment by awaisdar001** [comment]
Any plans for releasing this change? Bitly v2 in the current release is expiring soon!

**Comment by ellisonleao** [comment]
hey @awaisdar001 i am trying to finish the remaining tasks for next milestone. Should be pushed soon!

**Comment by awaisdar001** [comment]
> hey @awaisdar001 i am trying to finish the remaining tasks for the next milestone. It should be pushed soon!

Best, Thank you, I hope it will be soon enough before the Bitlay V2 deprecation.

**Comment by ellisonleao** [comment]
@awaisdar001 1.0 is out! Make sure to update it on your side

**Comment by awaisdar001** [comment]
Thats a great news!

Thank you.

**Comment by awaisdar001** [comment]
Posted too early, I can't see v1.0 in under releases?
https://github.com/ellisonleao/pyshorteners/releases

**Comment by ellisonleao** [comment]
@awaisdar001 haven't done the releases tags yet, but its already on pypi

**Pull request merged**